app/src/main/java/org/sil/gatherwords/MainActivity.java

```diff
@@ -165,7 +169,86 @@ public void onClick(View v) {
                 }
             });
 
+            ImageButton deleteButton = convertView.findViewById(R.id.session_list_button_delete);
+            deleteButton.setOnClickListener(new View.OnClickListener() {
+                public void onClick(View v) {
+                    Context context = inflater.getContext();
+                    session.deletedAt = new Date();
+                    new DeleteSessionFromDB(context).execute(session);
+
+                }
+            });
+
             return convertView;
         }
     }
+
+    private static class DeleteSessionFromDB extends AsyncTask<Session, Void, List<Session>> {
+        private SessionDao sDAO;
+        private WeakReference<MainActivity> mainActivityRef;
+
+        DeleteSessionFromDB(Context context) {
+            sDAO = AppDatabase.get(context).sessionDao();
+            mainActivityRef = new WeakReference<>((MainActivity) context);
+        }
+
+        @Override
+        protected List<Session> doInBackground(Session... sessions) {
+            sDAO.updateSession(sessions);
+            return sDAO.getAll();
+        }
+
+        @Override
+        protected void onPostExecute(List<Session> sessions) {
+            MainActivity mainActivity = mainActivityRef.get();
+            if ( mainActivity != null ) {
+                SessionListAdapter sessionListAdapter = (SessionListAdapter) ((ListView) mainActivity.findViewById(R.id.session_list)).getAdapter();
+                sessionListAdapter.sessions = sessions;
+                sessionListAdapter.notifyDataSetChanged();
+                mainActivity.showUndoSnackbar();
+            }
+        }
+    }
+
+    private void showUndoSnackbar() {
+        Snackbar mySnackbar = Snackbar.make(findViewById(R.id.mainView),
+                R.string.session_delete, Snackbar.LENGTH_LONG);
+        mySnackbar.setAction(R.string.undo, new UndoListener() );
+        mySnackbar.show();
+    }
+
+    public class UndoListener implements View.OnClickListener{
```

Comment:
Minor, but nice formatting is nice; at least on new code try to match style.  For example, spacing here and on line 216.

Reply:
Fixed... if I understand the issue correctly.